pkg/pool-weighted/test/ManagedPoolSwapFees.test.ts

```diff
@@ -0,0 +1,175 @@
+import { expect } from 'chai';
+import { BigNumber, Contract } from 'ethers';
+import { WEEK, DAY, MINUTE, advanceToTimestamp, currentTimestamp } from '@balancer-labs/v2-helpers/src/time';
+import { fp } from '@balancer-labs/v2-helpers/src/numbers';
+import * as expectEvent from '@balancer-labs/v2-helpers/src/test/expectEvent';
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+
+describe('ManagedPoolSwapFees', function () {
```

Comment:
We're missing a test for the fact that start time is resolved.

Reply:
We're also not testing the gradual behavior. Not sure what the correct strategy should be here - surely we don't want to restest the entire gradual change behavior, but shouldn't we at least have one test where we start a change and query the fee after some time?

Reply:
> surely we don't want to retest the entire gradual change behaviour, but shouldn't we at least have one test where we start a change and query the fee after some time?

I'd argue that verifying the return values from `getGradualSwapFeeUpdateParams` is sufficient. The tests for `ManagedPoolStorageLib` show that we'll interpolate between these values if the start and end swap fees are different.

Reply:
> We're missing a test for the fact that start time is resolved.

True, I'll add this.